application/log_handlers/handlers.py

```
@@ -0,0 +1,190 @@
+    def _open_new_log_file(self) -> None:
+        """
+            Generate a new log filename, open the file for writing, and switch the handler's stream.
+
+            This method updates `self.current_log_file`, opens a new stream in append-binary mode,
+            and reinitializes the handler to use the new stream. It should be called when the
+            current log file is deleted or after rotation.
+        """
+        new_log_file = self._generate_log_filename()
+
+        try:
+            new_stream = open(new_log_file, "ab")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)